JavaScript/chapter02/2.1 - Remove Dups/solution.js

```diff
@@ -0,0 +1,87 @@
+class node{
+	
+	constructor(data){
+		
+		this.data=data;
+		this.next=null;
+
+	}
+}
+
+
+class LinkedList{
+	
+	constructor(){
+		this.head=null;
+	}
+
+//Function to insert a node in the singlely linkedlist.
+	insert(data){
+		
+		var newnode=new node(data);
+		if(this.head==null){
+			this.head=newnode;
+		}
+
+		else{
+			
+			var temp=this.head;
+			while(temp.next){
+				temp=temp.next;			
+			}		
+				
+			temp.next=newnode;
+		}	
+	}
+
+//Function to print the singlely linkedlist.
+	printList(){
+		var temp=this.head;
+		var list="";
+		while(temp){
+			list+=temp.data+"->";
+			temp=temp.next;		
+		}
+		list+="null";
+		console.log(list);
+	}
+
+	
+
+//Function to remove dupliicate nodes from the singlely linkedlist.
+	removeDuplicates(){
+	
+		var temp1,temp2;
```

Comment:
nowadays the use of `var` is discouraged due to the lack of block scoping and them being hoisted.

it is recommended that declarations use `const` if possible. otherwise, you should use `let` and declare things in the same line as they are assigned a value:

```javascript
let currentNode = this.head;
```

also, try to give more descriptive names to your variables so that when you go back and read the code in a few weeks/months, you know what they are.

Reply:
Thanks for the inputs, will add them in the code.